packages/experimental/Avatar/src/Avatar.tsx

```diff
@@ -0,0 +1,162 @@
+
+export type AvatarData = {
+  /**
+   * The primary text to create initials with (e.g. a name)
+   */
+  primaryText?: string;
+
+  /**
+   * The secondary text to create initials with if primary text is not provided (e.g. an email address)
+   */
+  secondaryText?: string;
+
+  /**
+   * The image to be displayed
+   */
+  image?: Image;
+
+  /**
+   * The color that represents this avatar.
+   * This color will override the initials view's background color.
+   * If the avatar view is configured to display a border, this will be the border's color.
+   * The colored border will not be displayed if a custom border image is provided.
+   */
+  color?: string;
+
+  /**
+   * Image to be used as border around the avatar. It will be used as a pattern image color,
+   * but It will be scaled to fit the avatar size. If set, the hasBorder initializer value will be ignored,
+   * since it's assumed that the client intends to have a custom border.
+   */
+  customBorderImage?: Image;
+
+  /**
+   * The avatar view's presence state.
+   * The presence state is only shown when the style is set to AvatarStyle.circle.
+   */
+  presence?: Presence;
+};
+
+export type AvatarTokens = {
+  /**
+   * Supported Avatar sizes
+   */
+  size?: Size;
+};
+
+export type AvatarProps = ViewProps & {
+  /**
+   * Supported Avatar sizes
+   */
+  size?: Size;
+
+  /**
+   * Background Color of initials
+   */
+  backgroundColor?: string;
+
+  /**
+   * Image to be used as border around the avatar. It will be used as a pattern image color,
+   * but It will be scaled to fit the avatar size. If set, the hasBorder initializer value will be ignored,
+   * since it's assumed that the client intends to have a custom border.
+   */
+  customBorderImage?: Image;
+
+  /**
+   * Shape of the AvatarView
+   * Circle is used to represent people
+   * Square is used to represent organizations or teams
+   */
+  avatarStyle?: AvatarStyle;
+
+  borderColor?: string;
+
+  /**
+   * The avatar view's presence state.
+   * The presence state is only shown when the style is set to AvatarStyle.circle.
+   */
+  presence?: Presence;
+
+  /**
+   * When true, the presence status border is opaque. Otherwise, it is transparent.
+   */
+  usesOpaquePresenceBorder?: boolean;
+
+  // TODO Map the viewProp AccessibilityLabel to this
+  // overrideAccessibilityLabel? : string;
+
+  /**
+   * Used when avatarView doesn't have image or can't generate initials string
+   */
+  preferredFallBackImageStyle?: string;
+
+  /**
+   * Set to true to enable the pointer interaction on the avatar view, false by default.
+   * TODO Figure out how to make iOS only
+   */
+  hasPointerInteraction?: boolean;
+
+  // TODO Figure out how to make this a flat list at the AvatarProps level,
+  // and map to the inner custom property in the native view
+  // https://github.com/microsoft/fluentui-react-native/pull/372#discussion_r474326015
+  avatarData?: AvatarData;
```

Comment:
Why not just add the contents here to the props directly? Also why not add them directly on the native side? It's easy to collect props and construct an object on either side. If they are all just props it makes it easy to work with them correctly and ensure shallow compares work as expected.

Reply:
@JasonVMo , my new plan (similar to what I did for Shimmer) is to expose the props on the native side instead of having the intermediate `AvatarData` object. So the next steps for this PR will involve a change to FluentUI Apple, then an update here once we consume it. I expect that to take a few days, and I'll return to this PR once I can integrate those changes =)

Reply:
And I lied. Changing the native API is perhaps not the right approach. For context, in Swift, creating an AvatarView would look like this:

```swift
let avatarView = AvatarView(avatarSize: .large, withBorder: false, style: .circle)

avatarView.setup(primaryText: person.displayName, secondaryText: person.emailAddress, image: nil)
// OR
let avatarData = AvatarData(primaryText: person.displayName, secondaryText: person.emailAddress)
avatarView.setup(avatar: avatarData)
```
`primaryText` and `secondaryText` are not public properties on the native avatarview, but are set via the `setup` method. Calling the `setup` method again also clears out the old values (I guess it wanted to be an "immutable" view based on the properties you set it up with).

Changing the native API to make primary/secondaryText/etc publicwould mean introducing an alternative to setup method, and perhaps confuse clients. @harrieshin and @rdeassis expressed concern with doing so.

Instead, I went for the flat JS API that maps to a more complex native API route. I *believe* I did it correctly with the memoCache to ensure we don't have unnecessary re-renders, but I'd love to verify that with someone from CXE.